packages/react-combobox/src/components/Option/useOption.tsx

```diff
@@ -30,25 +30,32 @@ function getValueString(value: string | undefined, children: React.ReactNode) {
  * @param ref - reference to root HTMLElement of Option
  */
 export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElement>): OptionState => {
-  const { id, fluentKey: key, disabled, value } = props;
+  const { disabled, value } = props;
+  const optionRef = React.useRef<HTMLElement>(null);
 
   // context values
-  const idBase = useContextSelector(ListboxContext, ctx => ctx.idBase);
   const multiselect = useContextSelector(ListboxContext, ctx => ctx.multiselect);
   const onOptionClick = useContextSelector(ListboxContext, ctx => ctx.onOptionClick);
   const registerOption = useContextSelector(ListboxContext, ctx => ctx.registerOption);
   const selectedOptions = useContextSelector(ListboxContext, ctx => ctx.selectedOptions);
```

Comment:
It's not related to this PR, but is a small bomb 💣 I discussed this with @GeoffCoxMSFT and `useContextSelector` works pretty similar to [`reselect`](https://github.com/reduxjs/reselect) - it will trigger a rerender once a value from the function is not referentially equal.

But, this case can be easily solved as I see:

```diff
-const selectedOptions = useContextSelector(ListboxContext, ctx => ctx.selectedOptions);
```
```diff
-const selected = !!optionValue && !!selectedOptions.find(option => option.value === optionValue);
```

```diff
+  const selected = useContextSelector(
+    ListboxContext,
+    ctx => !!optionValue && !!ctx.selectedOptions.find(option => option.value === optionValue),
+  );
```

(in the same way as it's already done with `active`)